Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptions.swift

```diff
@@ -4,120 +4,114 @@
 
 import Foundation
 
+public typealias ChatMessageLayoutOptions = Set<ChatMessageLayoutOption>
+
+public extension ChatMessageLayoutOptions {
+    var rawValue: String {
+        map(\.rawValue).joined(separator: "-")
+    }
+
+    mutating func remove(_ options: ChatMessageLayoutOptions) {
+        self = subtracting(options)
+    }
+
+    mutating func insert(_ options: ChatMessageLayoutOptions) {
+        options.forEach { self.insert($0) }
+    }
+
+    /// Returns all options the current option set consists of separated by `-` character.
+    var description: String {
+        Self.singleOptions
+            .compactMap { contains($0) ? $0.optionName : nil }
+            .joined(separator: "-")
+    }
+
+    static let singleOptions: [ChatMessageLayoutOption] = [
+        .flipped,
+        .bubble,
+        .continuousBubble,
+        .avatarSizePadding,
+        .avatar,
+        .timestamp,
+        .authorName,
+        .text,
+        .quotedMessage,
+        .threadInfo,
+        .errorIndicator,
+        .reactions,
+        .onlyVisibleForYouIndicator,
+        .centered
+    ]
+}
+
 /// Describes the layout of base message content view.
-public struct ChatMessageLayoutOptions: OptionSet, Hashable {
-    public let rawValue: Int
+public struct ChatMessageLayoutOption: RawRepresentable, Hashable, Equatable, ExpressibleByStringLiteral {
```

Comment:
ExpressibleByStringLiteral can confuse our clients. Maybe this should be internal

Reply:
I think it will make it easier for them to create a new option.

Instead of:
```swift
extension ChatMessageLayoutOption {
    static let likes = ChatMessageLayoutOption(rawValue: "likes")
}
```
They can write:
```swift
extension ChatMessageLayoutOption {
    static let likes: Self = "likes"
}
```